src/DayColumn.js

```diff
@@ -84,22 +83,45 @@ class DayColumn extends React.Component {
     this.slotMetrics = this.slotMetrics.update(nextProps)
   }
 
-  triggerTimeIndicatorUpdate() {
-    // Update the position of the time indicator every minute
-    this._timeIndicatorTimeout = window.setTimeout(() => {
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.isNow !== this.props.isNow) {
+      this.clearTimeIndicatorInterval()
+
+      if (this.props.isNow) {
+        this.setTimeIndicatorPositionUpdateInterval(
+          prevState.timeIndicatorPosition === this.state.timeIndicatorPosition
+        )
+      }
+    }
+  }
+
+  /**
+   * @param tail {Boolean} - whether `positionTimeIndicator` call should be
+   *   deferred or called upon setting interval (`true` - if deferred);
+   */
+  setTimeIndicatorPositionUpdateInterval(tail = false) {
+    if (!tail) {
+      this.positionTimeIndicator()
+    }
+
+    this._timeIndicatorInterval = window.setInterval(() => {
```

Comment:
please use a setTimeout instead of setInterval, which can get weird on long running timers